api/v1beta1/cloudstackcluster_webhook_test.go

```diff
@@ -14,198 +14,85 @@ See the License for the specific language governing permissions and
-				},
-				Spec: CloudStackClusterSpec{
-					IdentityRef: &CloudStackIdentityReference{
-						Kind: defaultIdentityRefKind,
-						Name: identitySecretName,
-					},
-					Zone:    zone,
-					Network: network,
-				},
-			}
-			Expect(k8sClient.Create(ctx, cloudStackCluster)).Should(Succeed())
-		})
+	var ctx context.Context
+	forbiddenRegex := "admission webhook.*denied the request.*Forbidden\\: %s"
+	requiredRegex := "admission webhook.*denied the request.*Required value\\: %s"
+
+	BeforeEach(func() { // Reset test vars to initial state.
+		ctx = context.Background()
+		dummies.SetDummyVars()                       // Reset cluster var.
+		_ = k8sClient.Delete(ctx, dummies.CSCluster) // Delete any remnants.
+		dummies.SetDummyVars()                       // Reset again since the k8s client can set this on delete.
 	})
 
-	Context("When creating a CloudStackCluster with missing Network attribute", func() {
-		It("Should be rejected by the validating webhooks", func() {
-			ctx := context.Background()
-			cloudStackCluster := &CloudStackCluster{
-				TypeMeta: metav1.TypeMeta{
-					APIVersion: apiVersion,
-					Kind:       clusterKind,
-				},
-				ObjectMeta: metav1.ObjectMeta{
-					Name:      clusterName,
-					Namespace: clusterNamespace,
-					UID:       clusterID,
-				},
-				Spec: CloudStackClusterSpec{
-					IdentityRef: &CloudStackIdentityReference{
-						Kind: defaultIdentityRefKind,
-						Name: identitySecretName,
-					},
-					Zone: zone,
-				},
-			}
-			Expect(k8sClient.Create(ctx, cloudStackCluster).Error()).Should(MatchRegexp("admission webhook.*denied the request.*Required value\\: Network"))
+	Context("When creating a CloudStackCluster", func() {
+		It("Should accept a CloudStackCluster with all attributes present", func() {
+			Ω(k8sClient.Create(ctx, dummies.CSCluster)).Should(Succeed())
 		})
-	})
 
-	Context("When creating a CloudStackCluster with missing Zone attribute", func() {
-		It("Should be rejected by the validating webhooks", func() {
-			ctx := context.Background()
-			cloudStackCluster := &CloudStackCluster{
-				TypeMeta: metav1.TypeMeta{
-					APIVersion: apiVersion,
-					Kind:       clusterKind,
-				},
-				ObjectMeta: metav1.ObjectMeta{
-					Name:      clusterName,
-					Namespace: clusterNamespace,
-					UID:       clusterID,
-				},
-				Spec: CloudStackClusterSpec{
-					IdentityRef: &CloudStackIdentityReference{
-						Kind: defaultIdentityRefKind,
-						Name: identitySecretName,
-					},
-					Network: network,
-				},
-			}
-			Expect(k8sClient.Create(ctx, cloudStackCluster).Error()).Should(MatchRegexp("admission webhook.*denied the request.*Required value\\: Zone"))
+		It("Should reject a CloudStackCluster with missing Zones.Network attribute", func() {
+			dummies.CSCluster.Spec.Zones = []infrav1.Zone{{Name: "ZoneWNoNetwork"}}
+			Ω(k8sClient.Create(ctx, dummies.CSCluster)).Should(
+				MatchError(MatchRegexp(requiredRegex, "each Zone requires a Network specification")))
```

Comment:
Moved all of these tests to use
`Ω(<some func>).Should(MatchError(MatchRegexp(<regex>)))`

Instead of
`Ω(<some func>.Error()).Should(MatchRegexp(<regex>))`.

The old style would panic if some func returned nil.